src/lib/resolvers/guildThreadChannel.ts

```diff
@@ -0,0 +1,14 @@
+import type { Guild, ThreadChannel } from 'discord.js';
+import { Identifiers } from '../errors/Identifiers';
+import { err, ok, Result } from '../parsers/Result';
+import { resolveGuildChannel } from './guildChannel';
+
+export function resolveGuildThreadChannel(
+	parameter: string,
+	guild: Guild
+): Result<ThreadChannel, Identifiers.ArgumentGuildChannelError | Identifiers.ArgumentGuildThreadChannelError> {
+	const result = resolveGuildChannel(parameter, guild);
+	if (!result.success) return result;
+	if (result.value.isThread()) return ok(result.value);
+	return err(Identifiers.ArgumentGuildThreadChannelError);
+}
```

Comment:
Do you mean leave them in each file, but simplify the structure by calling this predicate function?
If yes I don't think it is really worth because it's only a few lines long.
Otherwise, I agree it would reduce a bit of duplication, but this would be there is not a 1-to-1 mapping from arguments to resolvers, which is kinda weird and not really what you would expect, don't you think?

Reply:
It's still cleaner than writing several lines of boilerplate in any case. I'm also not sure what you mean by losing a 1-1 mapping.

Reply:
> It's still cleaner than writing several lines of boilerplate in any case. I'm also not sure what you mean by losing a 1-1 mapping.

Now this functions will only be 1 line of code (calling the predicate with a custom callback). What I meant was, should we keep them in each file? But looking back to it this seems to be the best solution, so I'll do that.

Reply:
Added in [this commit](https://github.com/sapphiredev/framework/pull/237/commits/66d21f748f3bdcb268e0570592515dca40a1efc8), tell me what you think